src/main/java/picard/analysis/TheoreticalSensitivity.java

```diff
@@ -236,37 +237,30 @@ public TheoreticalSensitivity() {
      * @param sampleSize sampleSize is the total number of simulations to run
      * @param alleleFraction the allele fraction to evaluate sensitivity at
      * @param randomSeed random number seed to use for random number generator
-     * @return
+     * @return Theoretical sensitivity for the given arguments at a constant depth.
      */
     public static double sensitivityAtConstantDepth(final int depth, final double[] qualityDistribution, final double logOddsThreshold, final int sampleSize, final double alleleFraction, final int randomSeed) {
         final RouletteWheel qualityRW = new RouletteWheel(trimDistribution(qualityDistribution));
-        final Random uniformRNG = new Random(randomSeed);
-
-        final Random r = new Random(randomSeed);
-
+        final Random randomNumberGenerator = new Random(randomSeed);
         final BinomialDistribution bd = new BinomialDistribution(depth, alleleFraction);
 
-        double averageQuality = IntStream.range(0, qualityDistribution.length-1).mapToDouble(n -> n * qualityDistribution[n]).sum();
-        double var = IntStream.range(0, qualityDistribution.length-1).mapToDouble(n -> qualityDistribution[n] * Math.pow(n - averageQuality, 2.0)).sum() / (qualityDistribution.length - 1);
-//        System.out.println(qualityDistribution);
-//        System.out.println("averageQuality = " + averageQuality + "  var = " + var);
-
-//        IntStream.range(0, qualityDistribution.length-1).forEach((n) -> {
-//            System.out.println(n + " " + qualityDistribution[n]);
-//        });
+        // Calculate mean and deviation of quality score distribution to enable Gaussian sampling below.
+        final double averageQuality = IntStream.range(0, qualityDistribution.length-1).mapToDouble(n -> n * qualityDistribution[n]).sum();
+        final double qualityScoreVariance = IntStream.range(0, qualityDistribution.length-1).mapToDouble(n -> qualityDistribution[n] * Math.pow(n - averageQuality, 2.0)).sum() / (qualityDistribution.length - 1);
 
         int calledVariants = 0;
         for (int k = 0; k < sampleSize; k++) {
-            final int altDepth = bd.inverseCumulativeProbability(uniformRNG.nextDouble());
+            final int altDepth = bd.inverseCumulativeProbability(randomNumberGenerator.nextDouble());
 
-            int sumOfQualities;
-            if(altDepth < 10) {
+            final int sumOfQualities;
+            if (altDepth < largeNumberOfDraws) {
+                // If the number of alt reads is "small" we draw from the actual base quality distribution.
                 sumOfQualities = IntStream.range(0, altDepth).map(n -> qualityRW.draw()).sum();
+            } else {
+                // If the number of alt reads is "large" we draw from a Gaussian approximation of the base
+                // quality distribution to speed up the code.
+                sumOfQualities = (int) (averageQuality * altDepth + randomNumberGenerator.nextGaussian() * qualityScoreVariance * altDepth);
```

Comment:
hmmm. is this right? one draw with the same variance as the entire data? that seems wrong.